src/test/java/E2E/Pages/ProductsPage.java

```diff
@@ -1,18 +1,112 @@
+    WebElement removeSauceLabsBackpack;
+    @FindBy(xpath = "//*[@id='add-to-cart-sauce-labs-bike-light']")
+    WebElement addSauceLabsBikeLight;
+    @FindBy(xpath = "//*[@id='remove-sauce-labs-bike-light']")
+    WebElement removeSauceLabsBikeLight;
+    @FindBy(xpath = "//*[@id='add-to-cart-sauce-labs-bolt-t-shirt']")
+    WebElement addSauceLabsBoltT_Shirt;
+    @FindBy(xpath = "//*[@id='remove-sauce-labs-bolt-t-shirt']")
+    WebElement removeSauceLabsBoltT_Shirt;
+    @FindBy(xpath = "//*[@id='add-to-cart-sauce-labs-fleece-jacket']")
+    WebElement addSauceLabsFleeceJacket;
+    @FindBy(xpath = "//*[@id='remove-sauce-labs-fleece-jacket']")
+    WebElement removeSauceLabsFleeceJacket;
+    @FindBy(xpath = "//*[@id='add-to-cart-sauce-labs-onesie']")
+    WebElement addSauceLabsOnesie;
+    @FindBy(xpath = "//*[@id='remove-sauce-labs-onesie']")
+    WebElement removeSauceLabsOnesie;
+    @FindBy(xpath = "//*[@id='add-to-cart-test.allthethings()-t-shirt-(red)']")
+    WebElement addTest_allTheThings_T_Shirt_Red;
+    @FindBy(xpath = "//*[@id='remove-test.allthethings()-t-shirt-(red)']")
+    WebElement removeTest_allTheThings_T_Shirt_Red;
+    public void waitForLoading() {
+        getWait().forVisibility(header);
+        getWait().forVisibility(filterByName);
+        getWait().forVisibility(addSauceLabsBackpack);
+        getWait().forVisibility(addSauceLabsBikeLight);
+        getWait().forVisibility(addSauceLabsBoltT_Shirt);
+        getWait().forVisibility(addTest_allTheThings_T_Shirt_Red);
+        getWait().forVisibility(addSauceLabsOnesie);
+        getWait().forVisibility(addSauceLabsFleeceJacket);
+        getWait().forClickable(cardButton);
+    }
+    public void waitForLoadingRemoveButton(){
+        getWait().forVisibility(removeSauceLabsBackpack);
+        getWait().forVisibility(removeSauceLabsBikeLight);
+        getWait().forVisibility(removeSauceLabsBoltT_Shirt);
+        getWait().forVisibility(removeSauceLabsOnesie);
+        getWait().forVisibility(removeTest_allTheThings_T_Shirt_Red);
+        getWait().forVisibility(removeSauceLabsFleeceJacket);
+    }
+    public List<WebElement> getAddedItems(){
+        return driver.findElements(By.xpath("//button[text()='Remove']"));
+    }
 
-    public void waitForLoading(){
-        isElementDisplayed(header);
+    public int getProductCount(){
+        return driver.findElements(By.xpath("//*[@id='inventory_container']//*[@class='inventory_item']")).size();
+    }
+    public void filterDropdownButton(){
+        filterByName.click();
+    }
+    public void selectByFilterName(String name){
+        getSelect(filterByName).selectByVisibleText(name);
+    }
+    public String getFilterByName() {
+        return getSelect(filterByName).getFirstSelectedOption().getText();
+    }
+    public YourCardPage openYourCardPageDialog(){
+        cardButton.click();
+        return new YourCardPage(driver);
+    }
+    public void addItem() throws InterruptedException {
```

Comment:
Using sleep after every method very bap practice
Need to rewrite this method without sleep

You can do same like in Phonebook project  method(openTab)
```
public void openTab(ContactInfoTabs tab){
        driver.findElement(By.xpath("//*[@id='ngb-nav-"+tab.value+"']")).click();
    }
```
You can add enum with values which have title every product, and add in dynamic selector like in method openTab